scheduler/recovery.go

```diff
@@ -2,21 +2,64 @@ package scheduler
 
 import (
 	"context"
+	"errors"
 	"reflect"
+	"time"
+
+	"github.com/puppetlabs/horsehead/v2/netutil"
+)
+
+const (
+	defaultBackoffMultiplier         = time.Millisecond * 5
+	defaultMaxRetries                = 10
+	defaultResetRetriesTimerDuration = time.Second * 10
 )
 
+// ErrMaxRetriesReached is the error returned by RecoveryDescriptor if the max retries
+// have been reached.
+var ErrMaxRetriesReached = errors.New("RecoveryDescriptor: max retries have been reached")
+
+// RecoveryDescriptorOptions contains fields that allow backoff and retry parameters
+// to be set.
+type RecoveryDescriptorOptions struct {
+	// BackoffMultiplier is the timing multiplier between attempts using netutil.Backoff.
+	BackoffMultiplier time.Duration
+	// MaxRetries is the max times the RecoveryDescriptor should attempt to run the delegate
+	// descriptor during a reset retries duration.
+	MaxRetries int
+	// ResetRetriesTimerDuration is the time it takes to reset the retry count when running
+	// a delegate descriptor.
+	ResetRetriesTimerDuration time.Duration
```

Comment:
I find this name somewhat confusing since I feel like the timer is just an implementation detail here, maybe something like `ResetRetriesAfter` would be more intuitive?